pyinteraph/path_analysis.py

```diff
@@ -0,0 +1,577 @@
+import argparse
+import logging as log
+import numpy as np
+import networkx as nx
+import MDAnalysis as mda
+import itertools
+import re
+import matplotlib.pyplot as plt
+import seaborn as sns
+import matplotlib.colors as col
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+def build_graph(fname, pdb = None):
+    """Build a graph from the provided matrix"""
+
+    try:
+        data = np.loadtxt(fname)
+    except:
+        errstr = f"Could not load file {fname} or wrong file format."
+        raise ValueError(errstr)
```

Comment:
Actually, should I use a try except statement here at all? Isn't it possible that the pdb file could have some other type of error I'm not accounting for if I only use ValueError and some other error? Wouldn't it be better to just have the line `u = mda.Universe(pdb)` and let it crash if the pdb has some issue? Then it'll print out the whole error message and the user can figure out how to solve it.

Reply:
no - we should try to handle exceptions as completely as possible, it would be better not to make it crash. What I meant is that we can handle it in case it is a `OSError` (and give an appropriate error message - for file not found or not readable) or a more generic error for other exceptions. You can use the clause `except:` without specifying an exception class as a catch-all clause

Reply:
I think it should be fine now. I initially misunderstood and thought when you added an exception it replaced the original error message with the one you provide so I thought some information was being lost.